source/agora/consensus/Fee.d

```diff
@@ -238,7 +238,7 @@ public class FeeManager
         if (!minimumFee.mul(tx.sizeInBytes()))
             return "Fee: Transaction size overflows fee cap";
         if (tx_fee < minimumFee)
-            return "Transaction: Fee is less than adjusted minimum fee";
+            return "Transaction: Insufficient Implicit Fees";
```

Comment:
"Insufficient fee for transaction size"

I get what you mean by "implicit" fee (fees that are not explicit, which are those of the payload), but it is the main fee - what users will see the most. Hence qualifying it might be confusing for users, but to respect the goal of differentiating it from the payload fee, "for transaction" should do, what do you think ?

Reply:
BTW this is in the second commit, it shouldn't. In fact you could submit it as a PR to HEAD in its own commit if you want.

Reply:
I'll extract it into new PR and reword.

Reply:
> There's a lot of mixed up changes. It would be best to split the various fixes in their own commit / PR. And we probably will need to have the `TxBuilder` part merged before we actually enable the checks in consensus, so that Faucet starts to create valid txs, right ?

Yes I will split up the changes so we can get Faucet updated to be compatible first.